Sources/MessagingPush/MessagingPush.swift

```diff
@@ -58,23 +56,16 @@ public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, Messagi
      */
     @discardableResult
     @available(iOSApplicationExtension, unavailable)
-    public static func initialize(
-        configure configureHandler: ((inout MessagingPushConfigOptions) -> Void)? = nil
-    ) -> MessagingPushInstance {
-        if let configureHandler = configureHandler {
-            // pass current config reference to update it without needing to recreate
-            configureHandler(&moduleConfig)
-        }
-
-        let moduleInitializedFirstTime = shared.initializeModuleIfNotAlready()
+    public static func initialize(withConfig config: MessagingPushConfigOptions = MessagingPushConfigBuilder().build()) -> MessagingPushInstance {
+        let moduleInitializedFirstTime = shared.initializeModuleIfNotAlready(config: config)
 
         guard moduleInitializedFirstTime else {
             return shared
         }
 
         // Some part of the initialize is specific only to non-NSE targets.
         // Put those parts in this non-NSE initialize method.
-        if Self.moduleConfig.autoTrackPushEvents {
+        if config.autoTrackPushEvents {
             DIGraphShared.shared.automaticPushClickHandling.start()
```

Comment:
If we didn't have this code, we could probably merge `initialize` and `initializeForExtension`. But I think we can't merge them yet because of iOS restrictions for NSE.